ghost/core/core/server/services/collections/BookshelfCollectionsRepository.js

```diff
@@ -53,6 +53,26 @@ module.exports = class BookshelfCollectionsRepository {
         return this.#modelToCollection(model);
     }
 
+    /**
+     * NOTE: we are only fetching post_id column here to save memory on
+     *       instances with a large amount of posts
+     *
+     *       The method could be further optimized to fetch posts for
+     *       multiple collections at a time.
+     *
+     * @param {string} collectionId collection to fetch post ids for
+     * @param {Object} options bookshelf options
+     *
+     * @returns {Promise<Array<{post_id: string}>>}
+     */
+    async #fetchCollectionPostIds(collectionId, options = {}) {
+        return await this.#relationModel
+            .query()
+            .select('post_id')
+            .where('collection_id', collectionId)
+            .transacting(options.transaction);
```

Comment:
I think this needs to go in a conditional based on if `options.transaction` is present